control-plane/agents/src/bin/core/node/wrapper.rs

```diff
@@ -1393,6 +1393,18 @@ impl ReplicaApi for Arc<tokio::sync::RwLock<NodeWrapper>> {
         self.update_replica_states(ctx.deref_mut()).await?;
         Ok(local_uri)
     }
+
+    /// Set entity id for a given replica via grpc.
+    async fn set_replica_entity_id(
+        &self,
+        request: &SetReplicaEntityId,
+    ) -> Result<Replica, SvcError> {
+        let dataplane = self.grpc_client_locked(request.id()).await?;
+        let replica = dataplane.set_replica_entity_id(request).await?;
+        let mut ctx = dataplane.reconnect(GETS_TIMEOUT).await?;
+        self.update_replica_states(ctx.deref_mut()).await?;
```

Comment:
Doesn't seem right.. should only update this replica

Reply:
This refreshes resourceMap of the Nodewrapper right? We probably dont need this step at all.We will update replica in the registry anyway if rpc call is Ok(). let nodewrapper update happen at its interval?

Reply:
It refreshes everything which is not needed, only need to update the replica which gets returned.